src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/PrisonerInvolvement.kt

```diff
@@ -52,4 +63,34 @@ class PrisonerInvolvement(
     outcome = outcome,
     comment = comment,
   )
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+
+    other as PrisonerInvolvement
+
+    if (report != other.report) return false
+    if (id != other.id) return false
+    if (prisonerNumber != other.prisonerNumber) return false
+    if (prisonerRole != other.prisonerRole) return false
+    if (outcome != other.outcome) return false
+    if (comment != other.comment) return false
```

Comment:
I'm looking at your video and when you change the comment the record seemly is updated correctly.

My understanding is that things like `prisonerRole`, `outcome` and comment are now used to determine if it's the same record (`equals()`).

But does that mean that if prisoner `A11111AA` is involved as `PERPETRATOR` but then the value in the upsert request changes it to be `SUSPECTED_INVOLVED` they would be considered as two different entities and therefore a separate record would be created for `A11111AA`? (instead of updating the existing record)

It seems to work in your video so I am missing something here (but tbf I haven't finished watching it yet).

Reply:
I think I may start to get it.

Because we are not provided with the primary key in the upsert request, my understanding is that we'll delete the existing record if different and insert a new one.

So effectively anything which doesn't match would trigger an "update" (not in the DB sense but still the result is the same, old state is deleted, new state is saved)